### ISSUE TITLE
help -f is highlighting wrong

### ISSUE THREAD

**Issue by fdncred** [nushell/nushell#6443]
### Describe the bug

The `help -f in` command is highlighting items wrong now, only in the usage column. See screenshot.
![image](https://user-images.githubusercontent.com/343840/187246521-183db3a1-4777-4488-8446-059229acfb84.png)

### How to reproduce

help -f in

### Expected behavior

no unnecessary gaps between letters

### Configuration

|key|value|
|-|-|
|version|0.67.1|
|branch|main|
|commit_hash|3cb9147f22966b76549db38aab521e081c0c6ef6|
|build_os|windows-x86_64|
|build_target|x86_64-pc-windows-msvc|
|rust_version|rustc 1.61.0 (fe5b13d68 2022-05-18)|
|rust_channel|1.61.0-x86_64-pc-windows-msvc|
|cargo_version|cargo 1.61.0 (a028ae42f 2022-04-29)|
|pkg_version|0.67.1|
|build_time|2022-08-29 08:57:50 -05:00|
|build_rust_channel|release|
|features|database, dataframe, default, trash, which, zip|
|installed_plugins|custom-value generate, custom-value generate2, custom-value update, from parquet, gstat, inc, nu-example-1, nu-example-2, nu-example-3, query, query json, query web, query xml|

**Comment by fdncred** [comment]
@panicbit can you help me figure this out please? Something weird is going on. I think it's in table. I added this to help.rs and everything is highlighted correctly, but when it gets drawn in the table it's messed up. I'm not sure if it's related to your recent changes or if it's something caused by another problem.

```rust
                cols.push("usage".into());
                vals.push(Value::String {
                    val: if usage_match {
                        eprintln!("before usage highlighting: {}", &usage);
                        let usage =
                            highlight_search_string(&usage, &org_search_string, string_style)?;
                        eprintln!("after usage highlighting : {}", &usage);
                        usage
                    } else {
                        usage
                    },
                    span: head,
                });
```

![image](https://user-images.githubusercontent.com/343840/187252856-ebc92f83-6ced-4ec8-9240-a1c684a912b5.png)

**Comment by panicbit** [comment]
It's definitely not related to my changes. I've checked against a sufficiently old commit. I'm checking with a bisect how this got introduced (tho I fear this might have been when the feature initially got implemented 😅).

**Comment by panicbit** [comment]
Actually, seems like I might have just been running out of disk space... so maybe it's not actually related to `tabled`. I'll look into what might be causing this issue using my brain now 🤡.

**Comment by panicbit** [comment]
`help commands | find in` seems to show the same issue

**Comment by fdncred** [comment]
I tried the same thing. I think they both use the `highlight_search_string()` function so that would make sense. But what I can't get over is why is it only messing up the `usage` column?

**Comment by panicbit** [comment]
I think it's actually related to `tabled`: Try piping into `to json` or get an an individual affected record to show. The output will be correct!

**Comment by panicbit** [comment]
Or simply: `help commands | find in | get usage`

**Comment by panicbit** [comment]
My final proof:
![grafik](https://user-images.githubusercontent.com/628445/187276018-ba40001b-193e-427c-a4b9-8505e97db99e.png)

**Comment by fdncred** [comment]
nice research @panicbit! thanks so much for your willingness to jump in and help figure out what's happening - or at least point the finger. LOL. 😆  let's see what @zhiburt has to say.

**Comment by zhiburt** [comment]
From what I see it's very likely related to tabled.
See how it split *input* word into 2 😥

**Comment by zhiburt** [comment]
I've found the cause.
Already fixed, but I need to do a bit more testing.

ref: https://github.com/zhiburt/tabled/pull/207

**Comment by fdncred** [comment]
nice job @zhiburt!!

**Comment by zhiburt** [comment]
So as I said this issue must be fixed, see.

was
![image](https://user-images.githubusercontent.com/20165848/187661224-67c96a03-c437-460c-886a-674b59b24388.png)

now
![image](https://user-images.githubusercontent.com/20165848/187661162-ebd8e339-2db6-43b9-a2e7-356dca7f1fed.png)

_________________________________

But......

I've noticed another issue, which could occurred when word is colored by different colors.
There would be not used space.

See this example when we want to split `Hello World` by width of 8 chars.

We would get
![image](https://user-images.githubusercontent.com/20165848/187667224-b66e7dab-23d8-4259-b744-769c9e914193.png)

But I was thinking maybe we must get this.

![image](https://user-images.githubusercontent.com/20165848/187667419-f70017bf-727b-49d8-a2f7-87073cf6fba1.png)

The thing is that this behavior is depends on the color.
If `World` colored by 2 colors you'll see the second variant.
If by 1 color or no color you'd see the first case.

And I spend these 2 days on trying to always do the 2nd case.
Which I seems to achieved.
But..........................................
The cost is pretty big.
I mean I didn't profile the function but it VERY likely got way worse from performance point of view.
And what's also important, the implementation requires more extensive ansi usage.

See;
When we split string `"this is a long sentence"` in old implementation we would get.

```
            "| \u{1b}[32m\u{1b}[40mthis is a long \u{1b}[39m\u{1b}[49m   |"
            "| \u{1b}[32m\u{1b}[40msentence\u{1b}[39m\u{1b}[49m          |"
```

In a new we would get

```
            "| \u{1b}[32;40mthis\u{1b}[39m\u{1b}[49m\u{1b}[32;40m \u{1b}[39m\u{1b}[49m\u{1b}[32;40mis\u{1b}[39m\u{1b}[49m\u{1b}[32;40m \u{1b}[39m\u{1b}[49m\u{1b}[32;40ma\u{1b}[39m\u{1b}[49m\u{1b}[32;40m \u{1b}[39m\u{1b}[49m\u{1b}[32;40mlong\u{1b}[39m\u{1b}[49m\u{1b}[32;40m \u{1b}[39m\u{1b}[49m   |"
            "| \u{1b}[32;40msentence\u{1b}[0m          |"
```

Which is quite significant.

So I guess I spend 2 days in a hole.
And it's better to stick with old approach.

What do you think?
@fdncred

**Comment by fdncred** [comment]
I'd rather it be accurate. 

I also see some wasteful/needless escapes in

```
"| \u{1b}[32;40mthis
    \u{1b}[39m\u{1b}[49m\u{1b}[32;40m <- space here
    \u{1b}[39m\u{1b}[49m\u{1b}[32;40mis
    \u{1b}[39m\u{1b}[49m\u{1b}[32;40m <- space here
    \u{1b}[39m\u{1b}[49m\u{1b}[32;40ma
    \u{1b}[39m\u{1b}[49m\u{1b}[32;40m <- space here
    \u{1b}[39m\u{1b}[49m\u{1b}[32;40mlong
    \u{1b}[39m\u{1b}[49m\u{1b}[32;40m <- space here
    \u{1b}[39m\u{1b}[49m   |"
            "| \u{1b}[32;40msentence\u{1b}[0m          |"
```

the `\u{1b}[39m\u{1b}[49m\u{1b}` in all this above it totally wasteful. It's setting the default foreground and background and then immediately changing it to `u{1b}[32;40m`. So, those 39's and 49's server no purpose at all and are never really used.

**Comment by zhiburt** [comment]
> I'd rather it be accurate.

I did switched back to it.

> the \u{1b}[39m\u{1b}[49m\u{1b} in all this above it totally wasteful. It's setting the default foreground and background and then immediately changing it to u{1b}[32;40m. So, those 39's and 49's server no purpose at all and are never really used.

I agree that's something better be fixed. That's why I brought it.
But yet currently it something which is not my glance on.

I've put it as a todo: if you don't mind https://github.com/zhiburt/tabled/issues/217

**Comment by fdncred** [comment]
fixed in #6286
